scripts/generate-version.ts

```diff
@@ -0,0 +1,33 @@
+import { execSync } from 'child_process'
+import fs from 'fs'
+
+async function main(): Promise<void> {
+  let version = 'Unkown'
+  // Get the latest Git tag
+  const latestTag = execSync('git describe --tags --abbrev=0').toString().trim()
+
+  // Get the commit hash associated with the latest tag
+  const tagCommitHash = execSync(`git rev-list -n 1 ${latestTag}`).toString().trim()
+
+  // Get the current commit hash
+  const currentCommitHash = execSync('git rev-parse HEAD').toString().trim()
+
+  // Check if the current commit hash is equal to the tag commit hash
+  if (currentCommitHash !== tagCommitHash) {
+    // The current commit hash is not associated with the latest tag, return the commit hash
+    console.log(`Current commit hash ${currentCommitHash} is not associated with the latest tag ${latestTag}`)
+    console.log(`Returning commit hash ${currentCommitHash}`)
+    version = currentCommitHash
+  } else {
+    // The current commit hash is associated with the latest tag, return the tag
+    console.log(`Current commit hash ${currentCommitHash} is associated with the latest tag ${latestTag}`)
+    console.log(`Returning tag ${latestTag}`)
+    version = latestTag
+  }
+
+  const result = `export const CURRENT_VERSION = '${version}'\n`
```

Comment:
this can also be resolved without extra files and without extra build steps like this @aleortega 

```typescript
require('catalyst-client/package.json').version
```

Reply:
oddish automatically adds the commit hash to the package.json, like the version

Reply:
https://unpkg.com/browse/catalyst-client@2.0.0-beta.4/package.json

`require('catalyst-client/package.json').gitHead`

Reply:
I analyzed this approach as well, but I think it won't work because we are not maintaining the correct version on `package.json`. I think this is something we need to address but is not in place at the moment.

Reply:
@aleortega I think what @menduz  is saying is the correct version will be included in the package.json (https://www.npmjs.com/package/dcl-catalyst-client?activeTab=code), I was not aware of that, but I think it would work for us

Reply:
Oh, I nice, I wasn't aware of that mapping, I'm changing the code for this approach, thanks!